pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -270,3 +270,38 @@ func getLabels() map[string]string {
 func isShootNamespace(namespace string) bool {
 	return strings.HasPrefix(namespace, v1beta1constants.TechnicalIDPrefix)
 }
+
+type kubeAPIServerService struct {
```

Comment:
Can we add this to a separate file and call it `externalNameService` or similar?

Reply:
`externalNameService` sound weird in my opinion as there is nothing really external about the service. It points to the internal IP of the cluster's kube-apiserver and is not accessible externally.

Reply:
It is renamed to `internalNameService` as discussed, because the service points to the internal kube-apiserver service.